test/declaration_test.ts

```diff
@@ -133,6 +133,17 @@ describe('interfaces', () => {
   });
 });
 
+describe('function types', () => {
```

Comment:
The interface does describe a function type and is isomorphic with `(a: string) => number`, but the syntactical form is an interface with a single call signature - which is what we should be using as terminology here, as we're doing syntax directed translation and mostly ignore the actual types.